Gems/Atom/RPI/Code/Source/RPI.Public/DynamicDraw/DynamicDrawContext.cpp

```diff
@@ -298,11 +298,17 @@ namespace AZ
 
         ShaderVariantId DynamicDrawContext::UseShaderVariant(const ShaderOptionList& optionAndValues)
         {
-            AZ_Assert(m_initialized && m_supportShaderVariants, "DynamicDrawContext is not initialized or unable to support shader variants. "
-                "Check if it was initialized with InitShaderWithVariant");
-
             ShaderVariantId variantId;
 
+            if (!m_initialized)
+            {
+                AZ_WarningOnce("DynamicDrawContext", false, "%s This function has been disabled because, most likely, the UIPass Pass doesn't exist.\n", __FUNCTION__);
+                return variantId;
+            }
+
+            AZ_Assert(m_supportShaderVariants, "DynamicDrawContext is not initialized or unable to support shader variants. "
+                "Check if it was initialized with InitShaderWithVariant");
```

Comment:
I think it's better to switch to warning once if m_supportShaderVariants is false. 
Also the "not initialized" part should be removed since the it returns early if it's not initialized.

Reply:
Fixed.